### PR TITLE
test(unit): test_validator.py + test_dispatcher.py — Issue #4

### LINKED ISSUE #4: [Jingsi] test_validator.py + test_dispatcher.py — Unit tests missing
## Description

`tests/unit/test_validator.py` and `tests/unit/test_dispatcher.py` are both empty placeholders.

Note: `test_lambda_a.py` contains some indirect validator coverage, but the dedicated files are still missing.

## Tasks

**test_validator.py**
- [ ] Valid input (python / java / javascript)
- [ ] Missing code / empty code / code not a string / code exceeds 1 MB
- [ ] Missing language / unsupported language (e.g. COBOL) / case-insensitive language
- [ ] Missing student_id / empty student_id
- [ ] `normalize()` strips whitespace and lowercases language

**test_dispatcher.py** (boto3 fully mocked — no real AWS calls)
- [ ] Happy path: writes to SQS + DynamoDB, returns a valid `scan_id`
- [ ] SQS send failure: verify DynamoDB state is handled correctly
- [ ] `scan_id` format validation (`scan-` prefix + 8-char hex)

## Assignee

@tyrahappy (Jingsi)

### PR DESCRIPTION
**Pull request by tyrahappy**
## Summary

- Implements `tests/unit/test_validator.py` — 29 pure unit tests for `lambda_a/validator.py`
- Implements `tests/unit/test_dispatcher.py` — 17 moto-backed unit tests for `lambda_a/dispatcher.py`
- Adds `tests/unit/conftest.py` — auto-configures `sys.path` for pytest discovery
- Adds `moto[sqs,dynamodb,s3]>=5.0.0` to `lambda_a/requirements.txt`

All 46 tests pass locally (`pytest tests/unit/test_validator.py tests/unit/test_dispatcher.py -v`).

## Test coverage

**test_validator.py (29 cases)**
| Area | Cases |
|------|-------|
| Happy path | all 8 supported languages, case-insensitive language |
| `code` field | missing, empty, whitespace-only, non-string, >1MB, exactly 1MB, unicode |
| `language` field | missing, empty, unsupported, error message lists valid options |
| `student_id` field | missing, empty, whitespace-only, non-string |
| Edge cases | empty body, None values, extra fields ignored |
| `normalize()` | strips code/student_id whitespace, lowercases language, exact field set |

**test_dispatcher.py (17 cases)**
| Area | Cases |
|------|-------|
| Return value | returns string, `scan-` prefix, uniqueness per call |
| DynamoDB record | record created, `status=PENDING`, language/student_id/scan_id/created_at stored |
| SQS message | message sent, contains all 4 fields, body is valid JSON |
| Failure paths | DynamoDB error propagates, SQS error propagates |

## Test plan
- [x] `pytest tests/unit/test_validator.py -v` → 29 passed
- [x] `pytest tests/unit/test_dispatcher.py -v` → 17 passed
- [x] No real AWS credentials needed (moto intercepts all calls)

Closes #4

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by sunnythreethree — COMMENTED** [review]
test

**Review by sunnythreethree — COMMENTED** [review]
test

**Review by sunnythreethree — COMMENTED** [review]
test

File: sast-platform/tests/unit/test_dispatcher.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
+"""
+test_dispatcher.py — Unit tests for lambda_a/dispatcher.py
+Jingsi Zhang | CS6620 Group 9
+
+Uses moto to mock AWS (SQS + DynamoDB) without real credentials.
```
Comment (sunnythreethree) [inline sast-platform/tests/unit/test_dispatcher.py]: test comment

**Review by sunnythreethree — COMMENTED** [review]
test

**Review by sunnythreethree — COMMENTED** [review]
test

**Review by sunnythreethree — COMMENTED** [review]
test

**Review by sunnythreethree — COMMENTED** [review]
test

**Review by sunnythreethree — COMMENTED** [review]
test

**Review by sunnythreethree — COMMENTED** [review]
test

**Review by beibei-ui — APPROVED** [review]
Comments are all test placeholders — no actual issues raised. LGTM, approving.

**Review by sunnythreethree — COMMENTED** [review]
test

**Review by sunnythreethree — COMMENTED** [review]
test

**Review by sunnythreethree — COMMENTED** [review]
test

**Review by sunnythreethree — COMMENTED** [review]
test

**Review by sunnythreethree — CHANGES_REQUESTED** [review]
Good coverage overall — happy paths, boundary values, error cases, and moto integration all look solid. Found **1 bug** (dead mock) and **3 minor issues** detailed as inline comments.
